### ISSUE TITLE
[Android] ListView itemTap not fired when using Ripple as itemTemplate

### ISSUE THREAD

**Issue by ADjenkov** [Akylas/nativescript-material-components#46]
Ripple somehow consumes the TouchEvent and prevents its propagation to parent control.

Steps to reproduce: 
1. Run 'demo' app for Android 
2. Tap on any of the ListView example titles

Actual: Does not navigate to example, since `itemTap` is not fired at all.
Expected: Navigate to the selected example

Example:
 If you setup XML like this:
```
    <GridLayout tap="{{ onTap }}">
         <mdr:Ripple>
                <Label text="ripple"/>
         </mdr:Ripple>
    </GridLayout>
```

`onTap` callback will never be called

Note sure why this happens, but I believe it's related to that how Ripple handles TouchEvent. 
Just for the test you can add `isPassThroughParentEnabled="true"` to the `<mdr:Ripple>` and navigate to examples properly

**Comment by YouSour** [comment]
i got this problem too , i'm working with RadListView :(

**Comment by farfromrefug** [comment]
@ADjenkov this is simply because Ripple needs touch events to show the ripple. No real solution around this.
You can simply handle the tap even on the ripple with something like this ``` tap="{{ onItemTap(item)}}"```

It might work better using the mixins (which provide ripple and elevation for any view). But i have to test it

**Comment by ADjenkov** [comment]
@farfromrefug yes you can catch the tap еvent on the Ripple, but the problem is that tap should be passed to its parent so scenarios like the ListView/RadListView `itemTap` to work.

**Comment by farfromrefug** [comment]
@ADjenkov It s not a "problem" it is a normal behavior. Same thing happens with a button.
itemTap is a special event in a sense.
I can't do anything about it really

**Comment by ADjenkov** [comment]
@farfromrefug yes exactly, Ripple now behaves like a Button and no itemTap will be fired if button is set as itemTemplate also. If that's the case and this is the desired behaviour then you should probably modify your `demo` app home-page.xml since it will never navigate to the examples (since `itemTap` will never be called for Android)

_Also, I find it strange to behave differently on Android and iOS, since iOS will execute both the Ripple effect and the itemTap navigation_

**Comment by farfromrefug** [comment]
@ADjenkov new release should be out with a fix!

**Comment by ADjenkov** [comment]
@farfromrefug I think removing `view.setClickable(this.isUserInteractionEnabled);` will disable ripple effect since it's really not drawn (as your comment says). I'm not sure why you have to explicitly set `setClickable` since it should be set to true for every `View` with isUserInteractionEnabled by default.

**Comment by farfromrefug** [comment]
@ADjenkov Ok so there is an issue with Android version. Mine are showing just fine.
Can you give infos on your platform?

**Comment by ADjenkov** [comment]
@farfromrefug I'm testing demo app `ripple` examples on Emulator Api 19,23 and 27. Tapping on anything inside demo/examples/ripple-fragment.xml does not show ripple.

**Comment by farfromrefug** [comment]
@ADjenkov  will run some test. I am using the vue demo in dev. Will need to see if there is any difference with demo app

**Comment by ADjenkov** [comment]
@farfromrefug I haven't tested demo-ng and demo-vue, only 'demo'
Btw I've made a branch with migrating these example apps to run with {N} 6

**Comment by farfromrefug** [comment]
@ADjenkov actually found the true reason of the issue and why it was working the way it was.
I released 2.3.19 with the fix. If you are interested i opened an issue in {N} related to this https://github.com/NativeScript/NativeScript/issues/7717

**Comment by farfromrefug** [comment]
@ADjenkov can you confirm it works now?
